dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -496,48 +474,82 @@ public void removeAttachedFile(WorkflowTask task, String fileInode) throws DotDa
 	 * is in
 	 */
 	@CloseDBIfOpened
-	public List<WorkflowAction> findAvailableActions(Contentlet contentlet, User user) throws DotDataException,
-	DotSecurityException {
+	public List<WorkflowAction> findAvailableActions(final Contentlet contentlet, final User user)
+			throws DotDataException, DotSecurityException {
 
-		if(contentlet == null || contentlet.getStructure() ==null){
+		if(contentlet == null || contentlet.getStructure() ==null) {
+
+			Logger.debug(this, "the Contentlet: " + contentlet + " or their structure could be null");
 			throw new DotStateException("content is null");
 		}
 
-		List<WorkflowAction> actions= new ArrayList<WorkflowAction>();
-		if("Host".equals(contentlet.getStructure().getVelocityVarName())){
-			return actions;
-		}
+		final ImmutableList.Builder<WorkflowAction> actions =
+				new ImmutableList.Builder<>();
 
-		boolean isNew  = !UtilMethods.isSet(contentlet.getInode());
-		boolean canLock = false;
-		String lockedUserId =  null;
-		try{
-			canLock = APILocator.getContentletAPI().canLock(contentlet, user);
-			lockedUserId =  APILocator.getVersionableAPI().getLockedBy(contentlet);
-		} catch(Exception e){
+		if(Host.HOST_VELOCITY_VAR_NAME.equals(contentlet.getStructure().getVelocityVarName())) {

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "getStructure"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)